pylearn2/sandbox/rnn/space/__init__.py

```diff
@@ -116,11 +110,11 @@ def _format_as_impl(self, is_numeric, batch, space):
         else:
             if isinstance(space, SequenceDataSpace):
                 if is_numeric:
-                    formatted_batch = np.transpose(np.asarray([
+                    formatted_batch = np.swapaxes(np.asarray([
                         self.space._format_as_impl(is_numeric, sample,
                                                    space.space)
-                        for sample in np.transpose(batch, (1, 0, 2))
-                    ]), (1, 0, 2))
+                        for sample in np.swapaxes(batch, 0, 1)
+                    ]), 0, 1)
```

Comment:
I don't think it is that simple if `space.space` does not have `b` as first dimension, for instance if it is `Conv2DSpace(axes='c01b')`.